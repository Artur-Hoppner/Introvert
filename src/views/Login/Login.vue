<template>
  <v-container max-width="200">
    <v-layout>
      <v-flex class="login-form text-xs-center">
        <div class="display-1 mb-3 ">
          <v-icon x-large>home</v-icon> My Privacy
        </div>
        <v-card>
          <v-card-text>
            <div class="subheading">
              <template>Log in to your account</template>
            </div>
            <v-form @keyup.enter.native="login">
              <v-text-field
                v-model="username"
                prepend-icon="person"
                label="Username"
              ></v-text-field>

              <v-text-field
                v-model="password"
                light="light"
                prepend-icon="lock"
                label="Password"
                type="password"
              ></v-text-field>

              <v-btn id="loginButt" @click="login">Sign in</v-btn>
            </v-form>
          </v-card-text>
        </v-card>
        <div class="d-flex pa-6">
          Don't have an account?
          <v-btn id="goToButton" class="sign-up-link" @click="goTo"
            >Sign up</v-btn
          >
        </div>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script>
import { mapActions } from 'vuex';
import { mapFields } from 'vuex-map-fields';

export default {
  name: 'Login',
  computed: {
    ...mapFields(['login.username', 'login.password'])
  },
  methods: {
    ...mapActions(['login']),
    goTo() {
      this.$router.push('/registrate');
    }
  }
};
</script>
<style lang="scss" scoped>
.sign-up-link {
  margin-top: -7px;
  margin-left: 10px;
}
</style>
