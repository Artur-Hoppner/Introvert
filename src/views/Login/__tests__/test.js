// Utilities
import { mount, createLocalVue } from '@vue/test-utils';
// Libraries
import Vue from 'vue';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import Vuetify from 'vuetify';
// Components
import Login from '@/views/Login/Login.vue';

const localVue = createLocalVue();
localVue.use(Vuex, VueRouter);

jest.mock('vuex-map-fields', () => ({
  mapFields: jest.fn()
}));

describe('/login', () => {
  let vuetify;
  beforeEach(() => {
    Vue.use(Vuetify);
    vuetify = new Vuetify();
  });
  test('Test id Login is rendered', () => {
    // Arrange;
    const wrapper = mount(Login, {
      localVue,
      vuetify
    });
    // Assert
    expect(wrapper.element).toMatchSnapshot();
  });

  test('It should activate login on click', () => {
    const actions = {
      login: jest.fn()
    };
    const store = new Vuex.Store({
      actions
    });

    // Arrange;
    const wrapper = mount(Login, {
      localVue,
      vuetify,
      store
    });

    const button = wrapper.find('.v-btn');
    wrapper.vm.$on('login');
    expect(actions.login).toHaveBeenCalledTimes(0);
    button.trigger('click');
    expect(actions.login).toHaveBeenCalledTimes(1);
  });
});
