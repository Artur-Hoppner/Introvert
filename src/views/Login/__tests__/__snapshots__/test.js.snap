// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/login Test if Login is rendered 1`] = `
<div
  class="container max-width"
>
  <div
    class="layout"
  >
    <div
      class="flex login-form text-xs-center"
    >
      <div
        class="display-1 mb-3 "
      >
        <i
          aria-hidden="true"
          class="v-icon notranslate material-icons theme--light"
          style="font-size: 40px;"
        >
          home
        </i>
         My Privacy
      
      </div>
       
      <div
        class="v-card v-sheet theme--light"
      >
        <div
          class="v-card__text"
        >
          <div
            class="subheading"
          >
            Log in to your account
          </div>
           
          <form
            class="v-form"
            novalidate="novalidate"
          >
            <div
              class="v-input theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate material-icons theme--light"
                  >
                    person
                  </i>
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label theme--light"
                      for="inputUserName"
                      style="left: 0px; position: absolute;"
                    >
                      Username
                    </label>
                    <input
                      id="inputUserName"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <transition-group-stub
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate material-icons theme--light"
                  >
                    lock
                  </i>
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label theme--light"
                      for="inputPassword"
                      style="left: 0px; position: absolute;"
                    >
                      Password
                    </label>
                    <input
                      id="inputPassword"
                      type="password"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <transition-group-stub
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <button
              class="v-btn v-btn--contained theme--light v-size--default"
              id="loginButt"
              type="button"
            >
              <span
                class="v-btn__content"
              >
                Sign in
              </span>
            </button>
          </form>
        </div>
      </div>
       
      <div
        class="d-flex pa-6"
      >
        
        Don't have an account?
        
        <button
          class="sign-up-link v-btn v-btn--contained theme--light v-size--default"
          id="goToButton"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            Sign up
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
